lib/ansible/plugins/inventory/yc_compute.py

```diff
@@ -0,0 +1,300 @@
+            sa_file_contents = self.get_option('service_account_contents')
+            if bool(sa_file_path) == bool(sa_file_contents):
+                raise AnsibleError('Either "service_account_file" or "service_account_contents" must be set '
+                                   'when auth_kind is set to "serviceaccountfile"')
+            if sa_file_path:
+                try:
+                    with open(sa_file_path, 'r') as f:
+                        sa_file_contents = f.read()
+                except Exception as e:
+                    raise AnsibleError('Error reading Service Account data from file: "{0}": {1}'
+                                       .format(sa_file_path, to_native(e)))
+            try:
+                sa = json.loads(sa_file_contents)
+            except Exception as e:
+                raise AnsibleError('Error reading Service Account data from JSON: {0}'.format(to_native(e)))
+            self.sdk = yandexcloud.SDK(interceptor=interceptor, service_account_key=sa)
+
+        elif auth_kind == 'oauth':
+            oauth_token = self.get_option('oauth_token')
+            if not oauth_token:
+                raise AnsibleError('oauth_token should be set')
+            self.sdk = yandexcloud.SDK(interceptor=interceptor, token=oauth_token)
+        else:
+            raise AnsibleError('Unknown value for auth_kind: {0}'.format(auth_kind))
+
+    def parse(self, inventory, loader, path, cache=True):
+        super(InventoryModule, self).parse(inventory, loader, path)
+        self._read_config_data(path)
+        self.init_sdk()
+        self.service = self.sdk.client(InstanceServiceStub)
+
+        cache_key = self.get_cache_key(path)
+        # cache may be True or False at this point to indicate if the inventory is being refreshed
+        # get the user's cache option too to see if we should save the cache if it is changing.
+        user_cache_setting = self.get_option('cache')
+        # read if the user has caching enabled and the cache isn't being refreshed.
+        attempt_to_read_cache = user_cache_setting and cache
+        # update if the user has caching enabled and the cache is being refreshed; update this value to True
+        # if the cache has expired below.
+        cache_needs_update = user_cache_setting and not cache
+        instances = None
+        # attempt to read the cache if inventory isn't being refreshed and the user has caching enabled.
+        if attempt_to_read_cache:
+            try:
+                instances = self._cache[cache_key]
+            except KeyError:
+                # This occurs if the cache_key is not in the cache or if the cache_key expired,
+                # so the cache needs to be updated.
+                cache_needs_update = True
+
+        if not instances:
+            instances = []
+            remote_filter = self.get_option('remote_filter')
+            for folder_id in (self.get_option('folders') or []):
```

Comment:
Since this is documented as required, if a user does not provide `folders` (which as you mentioned, could be []), this plugin will fail. Since the user has to provide the option this should just be:
```
            for folder_id in self.get_option('folders'):
```

Reply:
This is minor issue, but still. When `folders:` section is empty in yaml get_option() returns None and user see this trace (with -vvv):
```
[WARNING]:  * Failed to parse .../inventory/yc_compute.yaml with yc_compute plugin: 'NoneType' object is not iterable

  File "/home/stef/dev/ansible/lib/ansible/inventory/manager.py", line 287, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "/home/stef/dev/ansible/lib/ansible/plugins/inventory/yc_compute.py", line 295, in parse
    for folder_id in self.get_option('folders'):
```
`or []` protects from this.